app/pdf_utils/fill_sections.py

```diff
@@ -32,51 +42,78 @@ def reference(reference_id: str) -> NoEscape:
     return NoEscape(f"\\cite{{{bib_reference_name(reference_id)}}}")
 
 
-def yield_text(paragraph):
+def wrap_text(data: document.TextLeaf) -> NoEscape:
+    """
+    Wraps text item with Latex commands corresponding to the sibbling elements
+    of the text item. Allows for wrapping multiple formatting options.
+    ----
+    eg: if data looks like: {"bold": true, "italic": true, "text": "text to format" }
+    then `wrap_text(data)` will return: `\\bold{\\italic{text to format}}`
+
+    """
+    e = utils.escape_latex(data["text"])
+    # e = data["text"]
+
+    for option, command in TEXT_WRAPPERS.items():
+        if data.get(option) and e.strip(" ") != "":
+            e = command(e)
+
+    return NoEscape(e)
+
+
+def get_reference_ids(text_element: list):
     """
-    Work in Progress: Generator type that may be used to handle paragraph text logic
+    Checks for the "type" key with a value of "ref",
+    Appends the corresponding "refId" value to a new list or variable.
+    Formats the reference items with helper functions
     """
-    section_p_text = pydash.get(obj=paragraph, path="text")
-    yield section_p_text
+
+    ref_ids = [item["refId"] for item in text_element if item.get("type") == "ref"]
+
+    x = []
+    for item in ref_ids:
+        x.append(reference(item))
+
+    return x
 
 
 def get_paragraph_text(section_element):
     """
     Utility function used in generator.py to get text input by user
     """
-    for _indx, paragraph in enumerate(section_element["children"]):
+    for _indx, text_leaf in enumerate(section_element["children"]):
+
+        # check if the text_leaf has the path text
+        section_p_text = pydash.get(obj=text_leaf, path="text")
 
-        # check if the paragraph has the path text
-        # if so get text
-        section_p_text = pydash.get(obj=paragraph, path="text")
+        if len(section_element["children"]) > 1:
```

Comment:
We can remove this as `for child in section_element["children"]:` will do the same thing with an empty array. This way we can have 1 less nested code.